delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/driver/web/DeliveryController.java

```diff
@@ -0,0 +1,54 @@
+package nl.softwarestrijders.waiter.delivery.infrastructure.driver.web;
+
+import nl.softwarestrijders.waiter.delivery.core.application.DeliveryCommandHandler;
+import nl.softwarestrijders.waiter.delivery.core.application.DeliveryQueryHandler;
+import nl.softwarestrijders.waiter.delivery.core.domain.Delivery;
+import nl.softwarestrijders.waiter.delivery.infrastructure.driver.web.dto.DeliveryDto;
+import nl.softwarestrijders.waiter.delivery.infrastructure.driver.web.dto.DeliveryStatusDto;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/delivery")
+public class DeliveryController {
+    private final DeliveryCommandHandler commandHandler;
+    private final DeliveryQueryHandler queryHandler;
+
+    public DeliveryController(DeliveryCommandHandler commandHandler, DeliveryQueryHandler queryHandler) {
+        this.commandHandler = commandHandler;
+        this.queryHandler = queryHandler;
+    }
+
+    @GetMapping
+    public List<DeliveryDto> listAll() {
+        return this.toDto(this.queryHandler.handleGetAllDeliveries());
+    }
+
+    @GetMapping("/{id}")
+    public DeliveryDto findDeliveryById(@PathVariable UUID id) {
+        return this.toDto(this.queryHandler.handleGetDeliveryById(id));
+    }
+
+    @GetMapping("/status/{id}")
```

Comment:
```suggestion
    @GetMapping("{id}/retrieve-status")
```
you can check the slides about combining REST with CQRS